aiida/tools/importexport/archive/common.py

```diff
@@ -61,3 +78,384 @@ def detect_archive_type(in_path: str) -> str:
         'Unable to detect the input file format, it is neither a '
         'folder, tar file, nor a (possibly compressed) zip file.'
     )
+
+
+def read_file_in_zip(filepath: str, path: str) -> str:
```

Comment:
Should this also be called `safe_read_file_in_zip` then?

Reply:
I'd say no, because there is no specific safety features in the read logic (although I don't think there is anything to make it unsafe)